### PR TITLE
embeddings: add cybertron local embeddings

### LINKED ISSUE #494: embedding: support cybertron local embeddings
Using https://github.com/nlpodyssey/cybertron you can generate embeddings locally in pure Go code. Not all models are supported, but I've successfully tested it with `all-MiniLM-L6-v2`, `bge-small-en-v1.5` and `e5-small-v2` (also `base` and `large` variants, but they get exponentially slower)

As this is CPU-based the performance isn't stellar, I'm getting about 20-30 embeddings per second on my laptop. Nevertheless, there are no dependencies on C-code or on CUDA, so it's very easy to get it to work.

Would it make sense to add an adapter for this in langchaingo?

Some sample code:

```go
m, err := tasks.Load[textencoding.Interface](&tasks.Config{
    ModelsDir: "models",
    ModelName: "sentence-transformers/all-MiniLM-L6-v2",
})
if err != nil {
    panic(err)
}

result, err := m.Encode(ctx, text, int(bert.MeanPooling))
if err != nil {
    panic(err)
}

embedding := result.Vector.Normalize2().Data().F32()
```

Cybertron will automatically fetch the model from Huggingface and cache it locally in the `ModelsDir`. The last statement will give you an `[]float32` of the embedding.

### PR DESCRIPTION
**Pull request by corani**
Not sure if you want to take on the extra dependency on the `cybertron` module, but this

Resolves #494

### embeddings: add cybertron local embeddings

Added a new package 'cybertron' under 'embeddings' to run embedding
models locally. This includes the main 'cybertron.go' file, a test
file 'cybertron_test.go', and an 'options.go' file for providing
various options for the client.

### PR Checklist

- [X] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [X] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [X] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: embeddings/cybertron/cybertron.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	PoolingStrategy bert.PoolingStrategyType
+}
+
+var _ embeddings.EmbedderClient = (*Cybertron)(nil)
+
+func NewCybertron(opts ...Option) (*Cybertron, error) {
+	return applyOptions(opts...)
+}
+
+func (c *Cybertron) CreateEmbedding(ctx context.Context, texts []string) ([][]float32, error) {
```
Comment (tmc) [inline embeddings/cybertron/cybertron.go:25]: Can you document exported symbols?

Comment (corani) [reply embeddings/cybertron/cybertron.go:25]: This is an implementation of the EmbedderClient interface. I can document it, but saw that in other places this wasn't done either (e.g. here: https://github.com/tmc/langchaingo/blob/main/embeddings/huggingface/huggingface.go#L32)

**Comment by corani** [comment]
I've added a bit more document. As mentioned in #494, the examples require a tagged version so I'll create a follow-up to add the example once this is merged and we have a new version.

**Review by tmc — APPROVED** [review]
LGTM

**Comment by tmc** [comment]
Can you resolve the go.mod conflict please?

**Comment by corani** [comment]
Sure, I'll have a look once I'm in the office (in an hour or so)

**Comment by corani** [comment]
> Can you resolve the go.mod conflict please?

Done

**Pull request merged**